rclcpp/test/test_intra_process_manager.cpp

```diff
@@ -0,0 +1,748 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <rclcpp/macros.hpp>
+
+// Mock up publisher and subscription base to avoid needing an rmw impl.
```

Comment:
Yeah, it kind of sucks that you have to do this, but I don't see a better way besides providing an empty rmw implementation (which is probably a bad idea). Maybe the mock classes could at least be in a header only used by `rclcpp/test`, in case we want to re-use them?